### PR TITLE
add feature: vs code intellisense to `devicon.json`

### LINKED ISSUE #1481: [FEATURE REQUEST] Validate and add snippet to `devicon.json` in VS Code
### I have searched through the issues and didn't find my problem.

- [X] Confirm

### Problem

In `devicon.json`, we can sometimes miss a property or even get confused about the value that should be added.

### Possible Solution

I think many users use VS Code or GitHub Codespaces. Also I got inspired by [@BenSouchet comment](https://github.com/devicons/devicon/issues/988#issuecomment-1265112532) in order to create a validator and a snippet for a new entry to `devicon.json` in VS Code.

This is the [***link***](https://github.com/lunatic-fox/devicon/tree/vscode_intellisense) to a branch with this proposed feature.

## Feature overview
Once you start typing `new` on `devicon.json` the snippet suggestion is activated.
new-entry

Hitting `Enter` will create a template object with all required properties.
new-entry-json

The validator will be working, that's why some parts are underlined in yellow.
If you hover over some property it'll show a short description.
> *All descriptions were based on* [*Updating `devicon.json`*](https://github.com/devicons/devicon/wiki/Updating-%60devicon.json%60).
hover-properties

`versions.svg`, `versions.font`, `aliases[].base` and `aliases[].alias` show autocompletion suggestions with all possible icon versions.
auto-completion-suggestions

`aliases[].base` and `aliases[].alias` are suggested properties, because the objects of `aliases` are optional.
auto-completion-suggestions-properties

## Validation
* `name` - `^[0-9a-z-]+$`
* `altnames[]` - *none*
* `tags[]` - `tags-enum.json`
  > ***Needs to be standardized!***
* `versions`
  * `.svg[]` - *icon versions*
  * `.font[]` - *icon versions*
* `color` - `^#([0-9a-fA-F]{3}|[0-9a-fA-F]{6})$`
  > ***Can be standardized to six digits!***
* `aliases[]`
  * `.?base` - *icon versions*
  * `.?alias` - *icon versions*

### Additional information

I made `tags-enum.json` just by grouping all existent tags in `devicon.json` `develop` branch with Node.js.

I think in a strategy of breaking tags by `-`, `/` and `space` in order to be just a set of single words. But I need opinions on that!

### PR DESCRIPTION
**Pull request by lunatic-fox**
## Double check these details before you open a PR

- [x] PR does not match another non-stale PR currently opened

## Features

VS Code Intellisense to `devicon.json` provides these features:
- A snippet to create a new input object by typing `new`
- Validates
    - `name` structure
    - `color` structure
    - `version.svg[]`, `version.font[]`, `aliases.?base` and `aliases.?alias` completion
- Provides some description for the properties based on the documentation

**This PR closes #1481**

### CODE REVIEW

**Review by Snailedlt — CHANGES_REQUESTED** [review]
Looking great!
One of the features I'm most looking forwards too, since this will boost productivity so much!

As always, there's a few suggestions I'd like you to look at before we're ready for merging this. Great general structure though, definitely good work! 💯

File: .vscode/devicon.code-snippets [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "new entry": {
+    "scope": "json",
```
[the remark this answers is not in the document]
Comment (lunatic-fox) [reply .vscode/devicon.code-snippets:3]: Yeah, I was wishing just the same, too bad that's not possible. But I think your suggested feature #1484 solves that issue.

**Review by Snailedlt — APPROVED** [review]
Looking good! ✔️
Can't wait to get this into master, well done with the changes :)

**Comment by lunatic-fox** [comment]
@Panquesito7, please review this request. 🙂
Here's another reason: #1514

**Pull request merged**